python/tvm/script/parser.py

```diff
@@ -794,28 +812,54 @@ def visit_Name(self, node):
         symbol = self.context.lookup_symbol(name)
         if symbol is not None:
             return symbol
-        self.report_error("Unknown identifier %s" % name)
+        self.report_error(f"Unknown identifier {name}.", node.span)
+
+    def transform_TypeVar(self, node):
+        """Type variable visitor.
+
+        Equivalent to `transform_Var` but for types.
+        """
+        name = node.id.name
+        symbol = Registry.lookup(name)
+        if symbol is not None:
+            return symbol
+        symbol = self.context.lookup_symbol(name)
+        if symbol is not None:
+            return symbol
```

Comment:
An alternative way, that I think it is equivalent, with less branches and return statements in the same function. Feel free to keep your version if you want :)

```suggestion
        symbol = Registry.lookup(name) or self.context.lookup_symbol(name)
        if symbol is not None:
            return symbol
```